solution.cpp

```diff
@@ -22,113 +22,159 @@ using std::string;
 using std::ifstream;
 using std::ofstream;
 
-/**
- * Find the maximum unsigned integer in `infilename`.
- *
- * Arguments:
- * - `infilename`: A string representing the file path to open.
- *     - Note: Since the string will not be changed, we declare it `const`.
- *       Since strings are objects (and therefore larger than fundamental
- *       types), we avoid copying it by passing it by reference.
- *
- * Returns:
- * - The maximum integer, or `-1` on error.
- *     - Note: It would be better style in C++ to throw an exception, but we
- *       haven't learned about those yet.
- */
+ifstream infilename;
+ofstream outfilename;
```

Comment:
these probably should be global -- especially if you're passing the name as a string to the functions that will be using these.